### ISSUE TITLE
`current_tags': uninitialized constant ActiveSupport::TaggedLogging::Formatter::IsolatedExecutionState (NameError)

### ISSUE THREAD

**Issue by HoneyryderChuck** [soutaro/steep#466]
Hi,

Just the the aforementioned error today after updating deps to latest version, `steep` to 0.46. A more "complete" log:

```
/home/.bundle/ruby/3.0.0/gems/activesupport-7.0.0/lib/active_support/tagged_logging.rb:60:in `current_tags': uninitialized constant ActiveSupport::TaggedLogging::Formatter::IsolatedExecutionState (NameError)
        from /home/.bundle/ruby/3.0.0/gems/activesupport-7.0.0/lib/active_support/tagged_logging.rb:45:in `push_tags'
        from /home/.bundle/ruby/3.0.0/gems/activesupport-7.0.0/lib/active_support/tagged_logging.rb:95:in `push_tags'
        from home/.bundle/ruby/3.0.0/gems/steep-0.46.0/lib/steep.rb:146:in `block in new_logger'
        from <internal:kernel>:90:in `tap'
        from /home/.bundle/ruby/3.0.0/gems/steep-0.46.0/lib/steep.rb:145:in `new_logger'
        from /home/.bundle/ruby/3.0.0/gems/steep-0.46.0/lib/steep.rb:158:in `log_output='
        from /home/.bundle/ruby/3.0.0/gems/steep-0.46.0/lib/steep.rb:162:in `<module:Steep>'
```

This happens because `activesupport` is a direct dependency, and v7 has just been released, which broke logging integration.

I think the "quick fix" for `steep` should be to "cap" the upperbound version to `~> 6.0` or smth. Going forward, I'd like to challenge whether activesupport is providing that much value here, given that it's used only for logging (?), and maybe keeping steep's own logger shouldn't be that much burden and would make steep much more interoperable.

**Comment by mullermp** [comment]
We're also impacted by this.

**Comment by POPPIN-FUMI** [comment]
I am having the same problem

```
steep check
/home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.0/lib/active_support/tagged_logging.rb:60:in `current_tags': uninitialized constant ActiveSupport::TaggedLogging::Formatter::IsolatedExecutionState (NameError)

        IsolatedExecutionState[thread_key] ||= []
        ^^^^^^^^^^^^^^^^^^^^^^
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.0/lib/active_support/tagged_logging.rb:45:in `push_tags'
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.0/lib/active_support/tagged_logging.rb:95:in `push_tags'
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/steep-0.47.0/lib/steep.rb:146:in `block in new_logger'
        from <internal:kernel>:90:in `tap'
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/steep-0.47.0/lib/steep.rb:145:in `new_logger'
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/steep-0.47.0/lib/steep.rb:158:in `log_output='
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/steep-0.47.0/lib/steep.rb:162:in `<module:Steep>'
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/steep-0.47.0/lib/steep.rb:139:in `<top (required)>'
        from <internal:/home/fudo/.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/home/fudo/.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from /home/fudo/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/steep-0.47.0/exe/steep:7:in `<top (required)>'
        from /home/fudo/.rbenv/versions/3.1.0/bin/steep:25:in `load'
        from /home/fudo/.rbenv/versions/3.1.0/bin/steep:25:in `<main>'
```

Gemfile
```
source "https://rubygems.org"

gem "dotenv", "2.7.6"
gem "activesupport", "7.0.0"
gem "sinatra", "2.1.0"
gem "sinatra-activerecord", "2.0.25"
gem "steep", "0.47.0"
```

Command
```
bundle
rbs collection init
rbs collection install
rbs validates
steep check
```

**Comment by mullermp** [comment]
We've temporarily fixed this by adding to the Gemfile:

```
gem 'activesupport', '~> 6'
```

**Comment by HoneyryderChuck** [comment]
@soutaro can I ask you to have a look at this issue and linked PRs? This makes steep unusable with ruby 3.1.

**Comment by samwightt** [comment]
Really wish this issue was getting some attention. Steep is currently unusable on Ruby 3.0.0 because of this.

**Comment by soutaro** [comment]
@HoneyryderChuck and everyone on this issue.

Thank you for reporting this. I found this today. 🙇‍♂️

**Comment by soutaro** [comment]
Merged #484. Now working for the release of a new version.

**Comment by soutaro** [comment]
Released Steep [0.47.1](https://rubygems.org/gems/steep/versions/0.47.1) with a fix for this problem. I'm really sorry for taking so long time to fix (to find) this problem. 🙇‍♂️
